software/controller/lib/core/system_constants.h

```diff
@@ -0,0 +1,92 @@
+/* Copyright 2021, RespiraWorks
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// All constants related to the ventilator design should live here. These
+// include pin mappings, venturi geometry, ...
+
+// Reference abbreviations ([RM], [PCB], etc) are defined in hal/README.md
+
+#pragma once
+
+#include "gpio.h"
+#include "units.h"
+
+/**************************************************************************************
+ *                                ELECTRICAL CONSTANTS                                *
+ **************************************************************************************/
+
+// The analog sensor on the daughterboard is hooked up to PC0 (see [PCB]) aka ADC1_IN1 (see [DS])
+static constexpr GPIO::AdcChannel InterimBoardAnalogPressure{GPIO::Port::C, 0, 1};
```

Comment:
This, for example results in a sort of global instance. Not so bad in itself. Worse is that it already needs to know the memory address of the port as defined in `GPIO::Port`. if the latter were somehow just an abstract signifier without architecture implementation content, what we could have here is the mapping and **mapping** only. Which gets at some point translated into an object of the like we see here when such object is instatiated explicitly as member of a subsystem of HAL?